core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -360,9 +360,10 @@ protected void doSample() {
                     try {
                         // its a listed node, light connect to it...
                         logger.trace("connecting to listed node (light) [{}]", listedNode);
-                        transportService.connectToNodeLight(listedNode);
+                        transportService.connectToNodeLight(listedNode, pingTimeout, !ignoreClusterName);
```

Comment:
Another idea: could we add an assertion to whatever action the handshake is that the stream's versions are set to the current version's minimum compatible version? As in "all handshake actions are performed using the minimum compatibility version". It'd be a nice test and a nice reminder that you cannot change this *at all* without bumping the minimum compatibility version.